pipeline/authn/authenticator_cookie_session.go

```diff
@@ -171,13 +175,14 @@ func forwardRequestToSessionStore(r *http.Request, checkSessionURL string, prese
 		return nil, helper.ErrForbidden.WithReason(err.Error()).WithTrace(err)
 	}
 
+	body, err := ioutil.ReadAll(res.Body)
```

Comment:
I figured, if the body cannot be decoded, that would be a general error and it would be good to bubble up. The changes below it hopefully make reading the flow easier.

Reply:
It's a good point to move `ReadAll` from the if statement, because if we won't read the `res.Body` in the else statement, we may have a goroutine leak.

I've come across this problem working with the oathkeeper, but it seems that we also need to defer the `res.Body.Close()` to close the response body explicitly. What do you think about adding a deferred statement with the `Close` method on the line above?

Reply:
@ant1k9 yeah, maybe — @aeneasr thoughts?

Reply:
Yes please defer with body.Close!

Reply:
Fixed!